src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -930,7 +930,7 @@ public override object VisitField(FieldSymbol symbol, TypeCompilationState argum
                     var lazySemanticModel = body == null ? null : new Lazy<SemanticModel>(() =>
                     {
                         var syntax = body.Syntax;
-                        var semanticModel = (CSharpSemanticModel)_compilation.GetSemanticModel(syntax.SyntaxTree);
+                        var semanticModel = (CSharpSemanticModel)AnalyzerDriver.GetOrCreateCachedSemanticModel(syntax.SyntaxTree, _compilation, _cancellationToken);
```

Comment:
We found out from perf run done against Cyrus' PR that most allocations are happening through the below callstack:

```
microsoft.codeanalysis.features!Microsoft.CodeAnalysis.UseAutoProperty.AbstractUseAutoPropertyAnalyzer`4[System.__Canon,System.__Canon,System.__Canon,System.__Canon].GetGetterField(class Microsoft.CodeAnalysis.SemanticModel,class Microsoft.CodeAnalysis.IMethodSymbol,value class System.Threading.CancellationToken)
+ microsoft.codeanalysis.csharp.ni!CSharpSemanticModel.GetSymbolInfoCore
|+ microsoft.codeanalysis.csharp.ni!CSharpSemanticModel.GetSymbolInfoFromNode
| + microsoft.codeanalysis.csharp.ni!CSharpSemanticModel.GetSymbolInfo
|  + microsoft.codeanalysis.csharp.ni!SyntaxTreeSemanticModel.GetSymbolInfoWorker
|   + microsoft.codeanalysis.csharp.ni!SyntaxTreeSemanticModel.GetMemberModel
|   |+ microsoft.codeanalysis.csharp.ni!SyntaxTreeSemanticModel.GetOrAddModelIfContains
|   | + microsoft.codeanalysis.csharp.ni!SyntaxTreeSemanticModel.GetOrAddModel
|   |  + system.collections.immutable!ImmutableInterlocked.GetOrAdd
|   |  |+ microsoft.codeanalysis.csharp.ni!SyntaxTreeSemanticModel.CreateMemberModel
|   |  ||+ microsoft.codeanalysis.csharp.ni!SyntaxTreeSemanticModel.GetDeclaredSymbol
```

Even though the bound nodes are cached onto the freshly created semantic model, the member models are not cached here and hence any GetSymbolInfo query will cause us to recreate the member model for the node.